sphinx/ext/napoleon/__init__.py

```
@@ -378,6 +382,140 @@ def _process_docstring(app: Sphinx, what: str, name: str, obj: Any,
     lines[:] = result_lines[:]
 
 
+def _unwrap_callable(obj: Any) -> Any:
+    try:
+        return inspect.unwrap(obj)  # type: ignore[arg-type]
+    except Exception:
+        return obj
+
+
+def _load_module(module_name: Optional[str]) -> Optional[ModuleType]:
+    if not module_name:
+        return None
+    module = sys.modules.get(module_name)
+    if module is not None:
+        return module
+    try:
+        return importlib.import_module(module_name)
+    except Exception:
```

[major] Same concern here: swallowing every Exception coming out of importlib.import_module turns real import failures into a silent None. Please catch only the specific exception types we expect (e.g. ModuleNotFoundError/ImportError). Anything else should surface so we notice a broken module import.